File: C_Programmes/Stack_Array_Implementation.c
//Perform the operations push, pop, peek and display by implementing stack using arrays

#include<stdio.h>
#include<limits.h>
#define MAX INT_MAX
int stack[MAX];
int top = -1;

void push();
void pop();
void peek();
void display();

int main()
{

    int choice = 0;
    do
    {
        printf("Enter 1 for push\n2 for pop\n3 for peek\n4 for display\n5 to quit\n");
        scanf("%d",&choice);
        switch(choice)
        {
            case 1:
                push();
                break;
            case 2:
                pop();
                break;
            case 3:
                peek();
                break;
            case 4:
                display();
                break;
            case 5:
                printf("Quit\n");
                choice = 5;
                break;
            default:
                printf("Wrong entry\nRetry\n");
                break;
        }
    }
    while(choice!=5);
    return 0;
}

void push()
{
    if(top==MAX-1)
    {
        printf("Stack overflow\n");
        return;
    }
    else
    {
        int element;
        top+=1;
        printf("Enter the element you want to push into the stack\n");
        scanf("%d",&element);
        stack[top]=element;
        printf("The elemenet %d has been pushed into the stack\n",element);
        return;
    }
}

void pop()
{
    if(top==-1)
    {
        printf("Stack underflow\n");
        return;
    }
    else
    {
        int element;
        element = stack[top];
        top-=1;
        printf("The elemenet %d has been popped from the stack\n",element);
        return;
    }
}

void peek()
{
    if(top==-1)
    {
        printf("Stack underflow\n");
        return;
    }
    else
    {
        int element;
        element = stack[top];
        printf("The elemenet %d is at the top of the stack\n",element);
        return;
    }
}

void display()
{
    if(top==-1)
    {
        printf("Stack underflow\n");
        return;
    }
    else
    {
        for(int i = 0; i<=top; i++)
        {
            printf("%d ",stack[i]);
        }
        printf("\n");
        return;
    }
}

File: Python3_Programmes/Stack_Array_Implementation.py
import sys
stack = []
top = -1
MAX = sys.maxsize
def push():
    global stack
    global top
    global MAX
    if(top==MAX-1):
        print("Stack overflow")
        return
    else:
        top+=1
        print("Enter the element you want to push into the stack")
        element = int(input())
        stack.append(0)
        stack[top]=element
        print(f"The element {element} has been pushed into the stack")
        return
def pop():
    global stack
    global top
    global MAX
    if(top==-1):
        print("Stack underflow")
        return
    else:
        element = stack[top]
        stack.remove(element)
        top-=1
        print(f"The element {element} has been popped from the stack")
        return
def peek():
    if(top==-1):
        print("Stack underflow")
        return
    else:
        element = stack[top]
        print(f"The element {element} is at the top of the stack")
        return
    
def display():
    if(top==-1):
        print("Stack underflow")
        return
    else:
        for i in range(top+1):
            print(f"{stack[i]} ",end='')
        print("");
        return

def main():
    global stack
    global top
    choice = 0
    while choice!=5:
        print("Enter 1 for push\n2 for pop\n3 for peek\n4 for display\n5 to quit");
        choice = int(input())
        if(choice==1):
            push()
            continue
        elif(choice== 2):
            pop()
            continue
        elif(choice== 3):
            peek()
            continue
        elif(choice== 4):
            display()
            continue
        elif(choice== 5):
            print("Quit");
            choice = 5
            continue
        else:
            print("Wrong entry\nRetry\n")
            continue
    
        
         
if __name__=='__main__':
    main()